### ISSUE TITLE
Do you shift the output label?

### ISSUE THREAD

**Issue by gaopengpjlab** [tatsu-lab/stanford_alpaca#45]
From your training code, the output label and input label is the same. Where do you shift the output label? Will this happen automatically inside trainer?

**Comment by gaopengpjlab** [comment]
if labels is not None:
            # Shift so that tokens < n predict n
            shift_logits = logits[..., :-1, :].contiguous()
            shift_labels = labels[..., 1:].contiguous()
            # Flatten the tokens
            loss_fct = CrossEntropyLoss()
            loss = loss_fct(shift_logits.view(-1, self.config.vocab_size), shift_labels.view(-1))

**Comment by pGit1** [comment]
@gaopengpjlab 
I have the same question. Where did you find the code above? I dont see it anywhere? Is this handled in trainer? I know its "supposed" to be.

**Comment by arisohn** [comment]
https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py#L703
